source/state_representation/include/state_representation/trajectory/JointTrajectory.hpp

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include "state_representation/space/joint/JointState.hpp"
+#include "state_representation/trajectory/TrajectoryBase.hpp"
+
+namespace state_representation {
+
+/**
+ * @class JointTrajectoryPoint
+ * @brief Struct to represent a joint trajectory point
+ */
+struct JointTrajectoryPoint : public TrajectoryPoint {};
+
+/**
+ * @class JointTrajectory
+ * @brief Class to represent a trajectory of joint points and corresponding durations
+ */
+class JointTrajectory : public TrajectoryBase<JointTrajectoryPoint> {
+public:
+  /**
+   * @brief Constructor with name and reference frame provided
+   * @param name the name of the state
+   */
+  explicit JointTrajectory(const std::string& name = "");
```

Comment:
```suggestion
  explicit JointTrajectory();
```

Reply:
here I don't really see why we wouldn't support this. I understand we don't support the construction of an empty `JointState`, but here I think we should stay symmetric with the `Trajectory` classes. 

So, `TrajectoryBase` and `CartesianTrajectory` can both be constructed with a name, I think so should `JointTrajectory`

Reply:
Yes you are right, I was just looking for an empty constructor, which this is I guess. Do you think there is a difference between this constructor and having two constructors, one actually empty and one with the string argument that doesnt have a default argument?

Reply:
in this case I just saw it as extra lines of code that don't differ. We might be more flexible by separating those 2 in case we want to do something else in the empty constructor, but for the time being I don't see the reason to separate them really